bluetooth: host: Check bounds on num_completed_packets event

kozdra opened the pull request:
If an event with corrupted length arrives, the extra check makes the handler return early to avoid reading data out of bounds.

jori-nordic commented on subsys/bluetooth/host/hci_core.c:
```diff
@@ -427,6 +427,11 @@ static void hci_num_completed_packets(struct net_buf *buf)
 	LOG_DBG("num_handles %u", evt->num_handles);
 
+	if (sizeof(*evt) + sizeof(evt->h[0]) * evt->num_handles > buf->len) {
+		LOG_ERR("evt num_handles too large");
```
could you add the expected and actual values in the log message? Also I think you can move the one at L428 to be after your check.

@jhedberg is it okay to just return here? I would think that if we get this we should rather call it quits and assert. I would suggest using the CHECKIF macro at least.

jhedberg replied:
@jori-nordic I don't think this is catching any host bugs. It's only testing that the event parameters are consistent with the length of the event itself (which doesn't tell you anything about the host state). It does indicate that we have a buggy controller, so it's then a bigger question how we tread obviously incorrect or corrupted data. This would likely cause big problems in the host's operation, but generally asserts have been reserved for issues which indicate bugs in the local code, and don't want remote entities to be able to trigger asserts (potentially DoS attacks) at will. The question is do we treat the controller as something "local" or something "remote".

jori-nordic replied:
Yeah. This is a bigger problem than just this one case, as johan says it depends on how we treat the controller.
I'm fine with the changes, could you just apply my suggestion for logging the sizes?

kozdra committed:
bluetooth: host: Check bounds on num_completed_packets event

If an event with corrupted length arrives, the extra check makes the
handler return early to avoid reading data out of bounds.

Signed-off-by: Arkadiusz Kozdra <akozdra@antmicro.com>

kozdra commented:
I squashed the changes so that the history looks better.
